Client/src/Components/Graph/Graph.tsx

```diff
@@ -1,439 +1,116 @@
+      mergeDatasetPoints(dataset, data)
     }
-    setYToggle(!isYLog)
-  }
 
-  const handleSettingsClick = () => {
-    setSettingsToggle(!showSettings)
+    return data
   }
 
-  const handleXUpperBoundChange = (event) => {
-    const { value } = event.target
-    if ((value < extremeBoundaries.maxX)) {
-      setBoundaries({
-        ...boundaries,
-        maxX: (Math.ceil(extremeBoundaries.maxX))
-      })
+  const mergeDatasetPoints = (dataset: IGraphDatasetModel, data: any[]) => {
+    if (!dataset || dataset.isHidden) {
+      return
     }
-    else {
-      setBoundaries({
-        ...boundaries,
-        maxX: value
-      })
-    }
-  }
-  const handleXLowerBoundChange = (event) => {
-    const { value } = event.target
-    if ((value <= 0 && isXLog) || (value > extremeBoundaries.minX)) {
-      setBoundaries({
-        ...boundaries,
-        minX: (Math.floor(extremeBoundaries.minX))
-      })
-    }
-    else {
-      setBoundaries({
-        ...boundaries,
-        minX: value
-      })
+    //for all current dataset points:
+    for (let pointIndex = 0; pointIndex < dataset.points.length; pointIndex++) {
+      //create new point that has previous points info and this one's
+      const mergedPoint = { ...data[pointIndex], [`${dataset.id}x`]: dataset.points[pointIndex].x, [`${dataset.id}y`]: dataset.points[pointIndex].y }
+      data[pointIndex] = mergedPoint
     }
   }
 
-  const handleYUpperBoundChange = (event) => {
-    const { value } = event.target
-    if ((value < extremeBoundaries.maxY)) {
-      setBoundaries({
-        ...boundaries,
-        maxY: (Math.ceil(extremeBoundaries.maxY))
-      })
-    }
-    else {
-      setBoundaries({
-        ...boundaries,
-        maxY: value
-      })
-    }
+  const subscribeToChartEvents = (chart: am4charts.XYChart, xAxis: am4charts.Axis, yAxis: am4charts.Axis) => {
+    chart.events.on("ready", function () {
+      //todo zoom to values or indeces?
+      axes[0].zoomStartIndex &&
+        axes[0].zoomEndIndex &&
+        xAxis.zoomToIndexes(axes[0].zoomStartIndex, axes[0].zoomEndIndex)
+
+      axes[1].zoomStartIndex &&
+        axes[1].zoomEndIndex &&
+        yAxis.zoomToIndexes(axes[1].zoomStartIndex, axes[1].zoomEndIndex)
+    })
   }
-  const handleYLowerBoundChange = (event) => {
-    const { value } = event.target
-    if ((value <= 0 && isYLog) || (value > extremeBoundaries.minY)) {
-      setBoundaries({
-        ...boundaries,
-        minY: (Math.floor(extremeBoundaries.minY))
-      })
-    }
-    else {
-      setBoundaries({
-        ...boundaries,
-        minY: value
-      })
-    }
-  }
-
-  const active = [null, null, null, null]
 
-  const ref = React.useRef(null)
+  const updateAxis = (xAxis, yAxis) => {
+    xAxis.title.text = `${axes[0].variableName}, ${axes[0].units}`
+    xAxis.logarithmic = axes[0].logarithmic || false
+    xAxis.renderer.minGridDistance = 40
 
-  const getScale = (isLog, rangeFrom, rangeTo) => {
```

Comment:
We need to make sure to put this functionality back in the future

Reply:
ill put this back in the later sprint when we pay close attention to the units and stuff